src/components/AvatarGroup/AvatarGroupCounterTooltipContent.jsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import Clickable from "../Clickable/Clickable";
+import Flex from "../Flex/Flex";
+import Avatar from "../Avatar/Avatar";
+import cx from "classnames";
+import PropTypes from "prop-types";
+import styles from "./AvatarGroupCounterTooltipContent.module.scss";
+
+const AvatarGroupCounterTooltipContent = ({ avatars, type, className }) => {
+  const getTooltipContent = avatarProps => {
+    return avatarProps?.tooltipProps?.content || avatarProps?.ariaLabel;
+  };
+
+  avatars = avatars.map(avatar => ({ value: avatar.props }));
+  const displayAsGrid = !avatars.some(item => getTooltipContent(item.value));
```

Comment:
let's separate AvatarGroupCounterTooltipContent to 2 components for make it more clear:
AvatarGroupCounterTooltipContent and AvatarGroupCounterTooltipGridContent. (there is enough differences between these 2 components).
In the parent component use useMemo that go over all the avatars on the counter and return 2 
thins: isGrid (if some avatar does not include value) and array of all getTooltipContent
We are using use memo for not iterate all the avatars, only when the array instance reference changed.